custom_targets/custom_targets.json5

```diff
@@ -1,15 +1,89 @@
 {
+  // This target clones the existing NUCLEO_L452RE_P target without making
+  // any changes other than an extra define.
   "L452RE_SIMPLE": {
     "inherits": ["NUCLEO_L452RE_P"],
     "macros_add": ["MY_CUSTOM_DEFINE=5"],
     "device_name": "STM32L452RETx"
   },
+
+  // This target extends MCU_STM32L452xE and defines a custom pinmap.
   "L452RE_CUSTOM_PINMAP": {
     "inherits": ["MCU_STM32L452xE"],
     "device_name": "STM32L452RETx"
   },
+
+  // This target extends NUCLEO_L452RE_P and defines a custom clock configuration.
   "L452RE_CUSTOM_CLOCK": {
     "inherits": ["NUCLEO_L452RE_P"],
     "device_name": "STM32L452RETx"
+  },
+
+  // This target extends NUCLEO_L452RE_P but remaps the memory for a different chip part number.
+  // This assumes we are using the -C variant of the chip with half the flash space.
+  "L452RC_CUSTOM_MEMORY_MAP": {
+    "inherits": ["NUCLEO_L452RE_P"],
+
+    // When using a MCU name whose data is not included with Mbed, it is mandatory to provide a
+    // memory_banks section.  This provides information about the RAM and flash banks to Mbed.
+    // Also note that you should NOT specify a device_name, as that is only used for Mbed-internal targets
```

Comment:
maybe a note about where we should look to obtain this data

Reply:
OK I added a new section 3.2 here https://github.com/mbed-ce/mbed-os/wiki/Mbed-Memory-Bank-Information